### DISCUSSION TITLE
unitycatalog plans to replace hms?

### DISCUSSION THREAD

**Discussion by melin** [unitycatalog/unitycatalog#306] [Q&A]
unitycatalog plans to replace hms? spark engine reads metadata directly from unitycatalog?
Compatible with hms client

**Comment by jaceklaskowski** [comment]
Hive was a target of Spark SQL long long before UC (cf. https://jaceklaskowski.medium.com/why-is-spark-sql-so-obsessed-with-hive-after-just-a-single-day-with-hive-289e75fa6f2b dated as far as Apr 9, 2016). I think it was one of those dreams of Spark devs (rxin, Michael and the crew) to get rid of Hive all together, and when they were fairly close Hive 3 came out with transactional tables and some other cool features that made this wish go away. Hive is in so many corporate places that it's going to take years to replace it even with better and cooler software.

With that said, I don't think Unity Catalog is going to replace HMS. They'll likely co-exist and (given others who wanna have their voice heard in the catalog scene - https://www.snowflake.com/blog/polaris-catalog-open-source/) there'll be more. We've just getting started IMHO.

The Spark SQL engine does not even support UC yet. This is being worked on and just a tiny bit of all expected features are available (see https://books.japila.pl/unity-catalog-internals/spark-integration/).

Is UC going to be compatible with HMS? No idea. I would not be surprised it will sometime, but it needs more devs to implement all these requested features yet very few can do it IMHO (even though it's not that hard but requires some time and effort).

**Reply by dennyglee** [reply]
To add to @jaceklaskowski 's call out - HMS is an important interface that allows compatibility for many different frameworks and languages.  IMHO, we should create an HMS interface for UC but I'm not sure where this would be in the list of many priorities we have, eh?!
